ods_ci/tests/Tests/100__deploy/130__operators/140__serverless_operator/140__serverless_operator.robot

```diff
@@ -0,0 +1,116 @@
+    Check Status     oc get KnativeServing knative-serving -n ${KNATIVESERVING_NS} -o json | jq '.status.conditions[] | select(.type=="Ready") | .status'     KnativeServing    "True"    # robocop: disable
+    Is Resource Present     Gateway    knative-ingress-gateway     ${KNATIVESERVING_NS}
+    Is Resource Present     Gateway    knative-local-gateway     ${KNATIVESERVING_NS}
+    Is Resource Present     Service    knative-local-gateway     ${ISTIO_NS}
+    Is Resource Present     deployment    controller     ${KNATIVESERVING_NS}
+    Wait For Pods Numbers  2    namespace=${KNATIVESERVING_NS}
+    ...    label_selector=app.kubernetes.io/component=controller    timeout=120
+    ${pod_names}=    Get Pod Names    ${KNATIVESERVING_NS}    app.kubernetes.io/component=controller
+    Verify Containers Have Zero Restarts    ${pod_names}    ${KNATIVESERVING_NS}
+    ${podname}=    Get Pod Name   ${OPERATOR_NAMESPACE}    label_selector=name=rhods-operator
+    Check For Errors On Operator Logs    ${podname}    ${OPERATOR_NAMESPACE}
+    Check DSC Conditions    ${KNATIVESERVING_NS}    default-dsc
+
+
+*** Keywords ***
+Suite Setup
+    [Documentation]    Suite Setup
+    Set Library Search Order    SeleniumLibrary
+    RHOSi Setup
+
+Suite Teardown
+    [Documentation]    Suite Teardown
+    Close All Browsers
+    RHOSi Teardown
+
+Assign Vars According To Product
+    [Documentation]    Assign vars related to product
+    [Arguments]    ${product}
+    IF    "${product}" == "RHODS"
+        Set Suite Variable    ${OPERATOR_APPNAME}     Red Hat OpenShift AI
+        Set Suite Variable    ${OPERATOR_NAME}    Red Hat OpenShift AI
+        Set Suite Variable    ${OPERATOR_NAMESPACE}    redhat-ods-operator
+    ELSE IF    "${product}" == "ODH"
+        Set Suite Variable    ${OPERATOR_APPNAME}  Open Data Hub Operator
+        Set Suite Variable    ${OPERATOR_NAME}    Open Data Hub Operator
+        Set Suite Variable    ${OPERATOR_NAMESPACE}    openshift-operators
+    END
+
+Is Resource Present
+    [Documentation]    Check CR
+    [Arguments]       ${resource}     ${resource_name}    ${namespace}
+    ${rc}=     Run and Return Rc
+    ...  oc get ${resource} ${resource_name} -n ${namespace}
+    Should Be Equal    "${rc}"    "0"    msg=${resource} does not exist
+
+Check Status
+    [Documentation]    Check Resource Status
+    [Arguments]       ${oc_get}     ${resource}    ${expected_status}
+    ${status}=     Run
+    ...  ${oc_get}
+    Log    ${status}    console=True
+    Should Be Equal    ${status}    ${expected_status}   msg=${resource} is not in Ready status
+
+Check DSC Conditions
+    [Documentation]    Checks that all conditions Reconciled Succesfully
+    [Arguments]    ${namespace}    ${dsc_name}
+    ${rc}    ${out}=    Run And Return Rc And Output
+    ...    oc get DataScienceCluster ${dsc_name} -n ${namespace} -o jsonpath='{.status.conditions[].reason}'
```

Comment:
How exactly we check the condition here? :thinking:

Reply:
This is meant to be displayed as it is very prone to errors depending on the installation

Reply:
changed

Reply:
Just realized that we are checking the condition of just one component - then... shouldn't we want it to be in `ReconcileReady` state? It's just one component here, it should be ready and working.

Reply:
`Should Be Equal As Integers ${rc} ${0}` are you sure this is actually checking what we want? it would check that the the oc get command has worked without error (i.e., `status.conditions[].reason` exists in the DSC).

Reply:
changed the keyword to wait for Conditions on DSC to be Reconciled successfully or timeout